src/firstP.js

```diff
@@ -0,0 +1,53 @@
+import { map, length, equals } from 'ramda';
+import curry1 from 'ramda/src/internal/_curry1';
+
+import resolveP from './resolveP';
+
+/**
+ * Returns a promise that is fulfilled by the first of given promises to be fulfilled,
+ * or rejected with an array of rejection reasons if all of the given promises are rejected.
+ *
+ * @func firstP
+ * @memberOf RA
+ * @category Function
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.22.0|v2.22.0}
+ * @sig [Promise a] -> Promise a
+ * @aliases headP
+ * @param {Iterable.<*>} iterable An iterable object such as an Array or String
+ * @return {Promise} Returns a promise that is fulfilled by the first of given promises to be fulfilled,
+ * or rejected when all of given promises to be rejected.
+ * @see {@link RA.anyP|anyP}
+ * @example
+ *
+ * RA.firstP([
+ *   Promise.resolve(1),
+ *   2,
+ *   Promise.reject(3),
+ * ]); //=> Promise(1)
+ */
+const firstP = iterable => {
+  const errors = [];
+  const isEqualsIterableLength = equals(length(iterable));
```

Comment:
`hasEqualIterableLength` is more proper variable name here.

Reply:
If the iterable is custom iterable here, the R.length will return always `NaN`. You cannot assume length of iterable without iterating it.
```js
const customIterable = {
    *[Symbol.iterator]() {
        yield 1;
        yield 2;
        yield 3;
    },
};

R.length(customIterable);